userspace/libscap/engine/kmod/scap_kmod.c

```diff
@@ -214,6 +219,15 @@ int32_t scap_kmod_init(scap_t *handle, scap_open_args *oargs)
 	uint64_t api_version = 0;
 	uint64_t schema_version = 0;
 
+	if (params->kmod_name && params->kmod_name[0] != '\0')
+	{
+		engine.m_handle->m_name = strdup(params->kmod_name);
+	}
+	else
+	{
+		engine.m_handle->m_name = strdup(SCAP_KERNEL_MODULE_NAME);
```

Comment:
`SCAP_KERNEL_MODULE_NAME` was left as a fallback; i am not sure we want this; i am open to discussions.

Reply:
probably i would remove `SCAP_KERNEL_MODULE_NAME` at all using `scap` as a default name in sinsp cpp method

Reply:
@gnosek WDYT?

Reply:
> Maybe even sinsp should own the device name the kmod driver is opening?

But we might have consumers that only use scap though, that's why i added the logic as near to driver as possible.

Reply:
I suppose the question then becomes "is it okay to require these consumers to know the module name". Since scap doesn't load the module (IIRC) and the responsibility is on whoever ends up calling scap_open, I think it's fine but it's not a hill I'm willing to die on.

Reply:
Uhm that's a good question, but I think the answer could be yes, if you are a consumer of the kmod you should know its name...This approach gives us also more robustness we explicitly said which kernel module we want instead of relying on our troubling building phase...

Reply:
for what concern the char[] vs char* question, I think we should avoid as much as we can using malloc/free if not necessary

Reply:
> This approach gives us also more robustness we explicitly said which kernel module we want instead of relying on our troubling building phase

but it also increases the split we have between different builds of effectively the same code (e.g. sysdig_probe vs falco) 🤷‍♂️ 

> I think we should avoid as much as we can using malloc/free if not necessary

This is as far from the hot path as possible, I don't think we need to count cycles here ;) Still, I'm fine with whatever direction we choose.